server/tinylicious/src/runner.ts

```diff
@@ -34,37 +35,42 @@ export class TinyliciousRunner implements utils.IRunner {
         private readonly storage: IDocumentStorage,
         private readonly mongoManager: MongoManager,
         private readonly contentCollection: ICollection<any>,
-    ) {
-    }
+    ) {}
 
     public start(): Promise<void> {
         this.runningDeferred = new Deferred<void>();
 
-        // Create the HTTP server and attach alfred to it
-        const alfred = app.create(
-            this.config,
-            this.storage,
-            this.mongoManager);
-        alfred.set("port", this.port);
-
-        this.server = this.serverFactory.create(alfred);
-
-        const httpServer = this.server.httpServer;
-
-        configureWebSocketServices(
-            this.server.webSocketServer,
-            this.orderManager,
-            this.tenantManager,
-            this.storage,
-            this.contentCollection,
-            new TestClientManager(),
-            new DefaultMetricClient(),
-            winston);
-
-        // Listen on provided port, on all network interfaces.
-        httpServer.listen(this.port);
-        httpServer.on("error", (error) => this.onError(error));
-        httpServer.on("listening", () => this.onListening());
+        this.ensurePortIsFree()
```

Comment:
same here. i don't think the deferred is necessesary, and await and try catch is preferred over then/catch, as it keeps the control flow eaiser to understand and modify.